### PR TITLE
vtxoMinOffchainTxAmount clamped to dust when below it

### PR DESCRIPTION
**Pull request by bitcoin-coder-bob**
closes https://github.com/arkade-os/arkd/issues/846

I think we were just reporting back the wrong field?

  Before:
```                                                                                                                                                                                                  
VtxoMinAmount:        s.vtxoMinOffchainTxAmount                                                                                                                                                                                  
```
  After:
``` 
VtxoMinAmount:        s.vtxoMinSettlementAmount
```

A client reads `vtxoMinAmount` from `GetInfo`, constructs a `RegisterIntent` with an amount above that value, and gets rejected. The fix of reporting `vtxoMinSettlementAmount` directly solves this because that's the value `RegisterIntent` checks against.

Move setting `vtxoMinSettlement`, `vtxoMinOffchain`, and `utxoMin` into util function, `resolveMinAmounts` in `internal/core/application/utils.go` and added test for it in `internal/core/application/service_test.go` in a test called `TestResolveMinAmount`.

## Summary by CodeRabbit

* **Refactor**
  * Unified min-amount handling so validations, error messages, and public info use a single consolidated VtxoMinAmount derived from a dust-based resolution.

* **Tests**
  * Added tests covering min-amount resolution across dust thresholds and edge cases (negative/unset values, zeros, exact dust, values above dust).

### CODE REVIEW

File: internal/core/application/service.go [hunk 1]
```diff
@@ -963,14 +954,14 @@ func (s *service) SubmitOffchainTx(
...
+		if out.Value < s.vtxoMinAmount {
 			return nil, errors.AMOUNT_TOO_LOW.New(
 				"output #%d amount is lower than min vtxo amount: %d",
-				outIndex, s.vtxoMinOffchainTxAmount,
+				outIndex, s.vtxoMinAmount,
 			).WithMetadata(errors.AmountTooLowMetadata{
 				OutputIndex: outIndex,
-				Amount:      int(s.vtxoMinOffchainTxAmount),
-				MinAmount:   int(s.vtxoMinOffchainTxAmount),
+				Amount:      int(s.vtxoMinAmount),
```
Comment (altafan) [inline internal/core/application/service.go]: wrong amount?

```suggestion
				Amount:      int(out.Value),
```

Comment (bitcoin-coder-bob) [reply internal/core/application/service.go]: good catch

File: internal/core/application/service.go [hunk 2]
```diff
@@ -963,14 +954,14 @@ func (s *service) SubmitOffchainTx(
...
 				"output #%d amount is lower than min vtxo amount: %d",
-				outIndex, s.vtxoMinOffchainTxAmount,
+				outIndex, s.vtxoMinAmount,
 			).WithMetadata(errors.AmountTooLowMetadata{
 				OutputIndex: outIndex,
-				Amount:      int(s.vtxoMinOffchainTxAmount),
-				MinAmount:   int(s.vtxoMinOffchainTxAmount),
+				Amount:      int(out.Value),
+				MinAmount:   int(s.vtxoMinAmount),
 			})
```
[the remark this answers is not in the document]
Comment (bitcoin-coder-bob) [reply internal/core/application/service.go:987]: wouldnt we want: 
```
if out.Value < s.vtxoMinAmount && !script.IsSubDustScript(out.PkScript) {
```

dust is always 330, so i am unsure how `if out.Value >= int64(dust) && out.Value < s.vtxoMinAmount {` would evaluate to true

[the remark this answers is not in the document]
Comment (bitcoin-coder-bob) [reply internal/core/application/service.go:987]: done in: [2ff7250](https://github.com/arkade-os/arkd/pull/923/commits/2ff72508e351a82d6db58ecdd33312ae41429d23)

File: internal/core/application/service.go [hunk 3]
```diff
@@ -162,6 +160,14 @@ func NewService(
 		return nil, fmt.Errorf("failed to generate ephemeral key: %s", err)
 	}
 
+	dustAmount, err := wallet.GetDustAmount(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get dust amount: %s", err)
+	}
+	vtxoMinAmount, utxoMinAmount = resolveMinAmounts(
+		vtxoMinAmount, utxoMinAmount, int64(dustAmount),
+	)
```
Comment (altafan) [inline internal/core/application/service.go]: this must be dropped after merging #924. The changes made to Start() below are just enough

File: internal/core/application/service.go [hunk 4]
```diff
@@ -985,14 +982,14 @@ func (s *service) SubmitOffchainTx(
 				})
 			}
 		}
-		if out.Value < s.vtxoMinOffchainTxAmount {
+		if out.Value < s.vtxoMinAmount && !script.IsSubDustScript(out.PkScript) {
```
Comment (altafan) [inline internal/core/application/service.go]:
```suggestion
		if out.Value < s.vtxoMinAmount {
```

File: internal/core/application/service.go [hunk 5]
```diff
@@ -1889,14 +1886,14 @@ func (s *service) RegisterIntent(
 					})
 				}
 			}
-			if amount < uint64(s.vtxoMinSettlementAmount) {
+			if amount < uint64(s.vtxoMinAmount) {
```
Comment (altafan) [inline internal/core/application/service.go]: In Start(), we should take care of storing the dust amount we get at line https://github.com/arkade-os/arkd/blob/master/internal/core/application/service.go#L227 and use it here

```suggestion
			if amount < uint64(s.dustAmount) {
```

File: internal/core/application/service.go (hunk 4 above)
Comment (bitcoin-coder-bob) [reply internal/core/application/service.go]: @altafan https://github.com/arkade-os/arkd/pull/923#pullrequestreview-3876157462  do you disagree or agree with the AIs assessment?

Without this additional check sub-dust OP_RETURN outputs were rejected (at line: https://github.com/arkade-os/arkd/blob/eb83ea9fd70fb20e3002ee2007fdea3a72c62fad/internal/core/application/service.go#L979) before    
  ever reaching the OP_RETURN exemption (at line: https://github.com/arkade-os/arkd/blob/eb83ea9fd70fb20e3002ee2007fdea3a72c62fad/internal/core/application/service.go#L992). 

So should the flow be:
  1. skip the min-amount check for OP_RETURN scripts  (i.e. add back in the  `&& !script.IsSubDustScript(out.PkScript)`)                                                                                                                                          
  2. enforce that sub-dust outputs must be OP_RETURN scripts

File: internal/core/application/service.go (hunk 5 above)
Comment (bitcoin-coder-bob) [reply internal/core/application/service.go]: we want `RegisterIntent` to enforce the same minimum that `GetInfo` advertises:  `VtxoMinAmount: s.vtxoMinAmount` https://github.com/arkade-os/arkd/blob/eb83ea9fd70fb20e3002ee2007fdea3a72c62fad/internal/core/application/service.go#L2162 which `SubmitOffchainTx` enforces https://github.com/arkade-os/arkd/blob/eb83ea9fd70fb20e3002ee2007fdea3a72c62fad/internal/core/application/service.go#L979. Previously a client checking `GetInfo` would see one minimum but `RegisterIntent` would enforce a different one.

Comment (altafan) [reply internal/core/application/service.go]: yeah and that different one was set by default to dust amount if not specified in config. We decided to not have this value configurable basically and always allow to renew all vtxos that have amount at least dust (330 sats)

Comment (bitcoin-coder-bob) [reply internal/core/application/service.go]: Bit confused, seeking clarification. 

Since `vtxoMinAmount` will always equal dust (330), the check at line 979 (`out.Value < s.vtxoMinAmount`) rejects sub-dust outputs before they ever reach the OP_RETURN exemption at line 990. Should we:    
   
  1. Remove the `vtxoMinAmount` check at line 979 entirely (the dust check at line 990 already handles it with the OP_RETURN exemption), or                                                                  
  2. Can sub-dust OP_RETURN outputs never occur in `SubmitOffchainTx`, so this is fine as-is? 
  3. no action needed?

Comment (altafan) [reply internal/core/application/service.go]: `vtxoMinAmount` refers to the min allowed output amount in offchain tx. It's set to dust amount if only if it's config value is `-1`. So for example:
* if env var is unset -> min vtxo amount = -1 -> min vtxo amount = 330 -> sub-dust output are not allowed in offchain txs
* env var set to `5` -> min vtxo amount = 5 -> sub-dust outputs are allowed in offchain txs with min possible value of `5` sats
